### PR TITLE
Add `train_mace.py` for easier reproducibility

### PR DESCRIPTION
**Pull request by chiang-yuan**
This PR provides training script for reproducing universal mace checkpoint.

### CODE REVIEW

**Comment by janosh** [comment]
@chiang-yuan Thanks for this PR! I think it would be nice to have a single Python file for training MACE so I merged `mace-universal-distributed.sbatch` and `run_train.py` into `train_mace.py`.

In particular, it would be nice not to have any shell scripts since they're so much harder to read, test and fix if something breaks. Could you explain what `setup.sh` does? Is it strictly necessary or perhaps Perlmutter specific? It looks hard to convert that to Python so I left it as is for now.

Would be great if you could give the new `train_mace.py` a try to see if it still works. Since it now uses the `slurm_submit` utility from `matbench_discovery.slurm`, you should be able to launch it as a slurm job via:

```sh
python models/mace/train_mace.py slurm-submit
```

**Comment by chiang-yuan** [comment]
`setup.sh` is the NERSC provided script modified by me to allow variable-time job on preemptible queue so that the training can be requeued and rerun automatically. I assume this is perlmutter specific.

File: models/mace/json_to_extxyz.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+for material_id in tqdm(json_data):
+    for trajectory_id in json_data[material_id]:
+        out_xyz = f"{out_dir}/{material_id}.extxyz"
+
+        if os.path.isfile(out_xyz):
+            traj = read(out_xyz, index=":", format="extxyz")
+            combined.append(traj)
+            continue
+
+        block = copy.deepcopy(json_data[material_id][trajectory_id])
```
Comment (janosh) [inline models/mace/json_to_extxyz.py]: Do we actually need the `deepcopy` here?

**Comment by janosh** [comment]
> I assume this is perlmutter specific.

In that case we don't need to include `setup.sh` for repro purposes. I'll remove it. Then the only thing left here is to make sure the single-file `train_mace.py` actually works. Would be great if you could try it out with your local MACE branch and just make sure it starts training without errors.

**Comment by chiang-yuan** [comment]
Sure I can try it! Sorry for not being very fast about this. A lot of things going on on my side now... Will at least test and try this weekend.

**Comment by janosh** [comment]
You are fast! 🚀  This PR is much appreciated!

**Comment by chiang-yuan** [comment]
As of now, `mace/develop` does not have `HDF5Dataseet` implemented

```
ImportError: cannot import name 'HDF5Dataset' from 'mace.data'
```

File: models/mace/train_mace.py [hunk 2]
```diff
@@ -695,7 +701,7 @@ def main(*args: Any, **kwargs: Any) -> None:
         train_file="../train.h5",
         valid_file="../valid.h5",
         statistics_file="../statistics.json",
-        loss="uip",
+        loss="stress",
```
Comment (janosh) [inline models/mace/train_mace.py:706]: Is this accidental?

Comment (janosh) [reply models/mace/train_mace.py:706]: `"uip"` loss is not available in the MACE repo, only in @chiang-yuan's fork.

**Pull request merged**